plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -626,12 +641,43 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                                ((_produce_block_cpu_effort.count() / 1000) * config::producer_repetitions) );
    }
 
+   void update_active_finalizers() {
+      if (!_is_savanna_active || _producers.empty())
+         return;
+
+      auto& chain  = chain_plug->chain();
+      finalizer_policy_ptr fin_policy = chain.head_active_finalizer_policy();
+      assert(fin_policy);
+      _is_producer_active_finalizer = std::ranges::any_of(fin_policy->finalizers, [&](const auto& f) {
+         return _producers.contains(to_account_name_safe(f.description));
+      });
+      _other_active_finalizers = std::ranges::any_of(fin_policy->finalizers, [&](const auto& f) {
+         return !_producers.contains(to_account_name_safe(f.description));
+      });
+      if (!_is_producer_active_finalizer || !_other_active_finalizers) {
+         if (fin_policy = chain.head_pending_finalizer_policy(); fin_policy) {
+            if (!_is_producer_active_finalizer) {
```

Comment:
This section of code is the same as starting from line 651.

Reply:
Not clear how to make it much simplier.

Reply:
OK.